include/video.h

```diff
@@ -200,6 +200,21 @@ struct RenderParams {
 	//   - height = 200
 	//   - pixel_aspect_ratio = 5/6 (1:1.2)
 	VideoMode video_mode = {};
+
+	constexpr bool operator==(const RenderParams& that) const
+	{
+		return (width != that.width || height != that.height ||
+		        double_width != that.double_width ||
+		        double_height != that.double_height ||
+		        pixel_aspect_ratio != that.pixel_aspect_ratio ||
+		        pixel_format != that.pixel_format ||
+		        video_mode != that.video_mode);
+	}
```

Comment:
Nice improvement.  There's some code in video capture that can use this.

Reply:
Although wait... aren't these checks inverted?  Should be == and &&.

Reply:
If you're confident that each struct members can be compared with it's own `==` operator, you can use the compiler to generate the comparison operator:

```c++
struct MyStruct {
    int x;
    double y;

    bool operator==(const MyStruct& that) const = default;
};
```

This will create a body that runs `==` on each member against the other.

Reply:
Uhhuhh, good pickup. I guess things keep working but do potentially a lot more work after this change 😅 (it's used to only recreate/re-render stuff when there's a change)

That defaulting thing seems handy; I was looking for it but couldn't find it. 🚀

Reply:
Maybe another day @kcgen 😅 

```
warning: defaulted comparison operators are a C++20 extension [-Wc++20-extensions]
```

Reply:
Argh; darn. I was able to use a default `=` operator in:
```c++
MidiWork& operator=(MidiWork&&) = default;
Section& operator=(Section&& other) = default;
Section_line& operator=(Section_line&& other) = default
```

But I guess comparisons weren't ready for C++17 at the time :-)

Reply:
@kcgen Well, it goes back to C I think.  You could always use `=` to copy values from one struct to another but `==` was a compile error in plain C.  So I think default assignment operator would have been around long before C++ 17.

Reply:
Ahh indeed - Thanks for clarifying, @weirddan455 :+1: